asv_bench/benchmarks/reshape.py

```diff
@@ -111,7 +111,7 @@ def setup(self, dtype):
             values = np.take(list(string.ascii_letters), indices)
             values = [pd.Categorical(v) for v in values.T]
 
-        self.df = DataFrame(values, index, columns)
+        self.df = DataFrame({i: cat for i, cat in enumerate(values)}, index, columns)
```

Comment:
why is this needed? (only for int)? e.g. category seems to do this already

Reply:
ATM the constructor special-cases list-of-categorical, but that behavior is deprecated.  this is what we will want post-deprecation, and avoids a warning in the meantime